nucypher/blockchain/eth/actors.py

```diff
@@ -1244,6 +1244,28 @@ def disable_winding_down(self) -> TxReceipt:
         receipt = self._set_winding_down(value=False)
         return receipt
 
+    @property
+    def is_taking_snapshots(self) -> bool:
+        taking_snapshots = self.staking_agent.is_taking_snapshots(staker_address=self.checksum_address)
+        return taking_snapshots
+
+    @only_me
+    @save_receipt
+    def _set_snapshots(self, value: bool) -> TxReceipt:
+        # if self.is_contract:
```

Comment:
What are you thinking here?

Reply:
Placeholder for staking contract interaction, added TODOs